Day-8_Objects/objects.js

```diff
@@ -0,0 +1,417 @@
+      age: 20,
+      isLoggedIn: true,
+      points: 50
+    },
+    Thomas: {
+      email: 'thomas@thomas.com',
+      skills: ['HTML', 'CSS', 'JavaScript', 'React'],
+      age: 20,
+      isLoggedIn: false,
+      points: 40
+    },
+    Paul: {
+      email: 'paul@paul.com',
+      skills: ['HTML', 'CSS', 'JavaScript', 'MongoDB', 'Express', 'React', 'Node'],
+      age: 20,
+      isLoggedIn: false,
+      points: 40
+    }
+  }
+
+
+// Find the person who has many skills in the users object.
+let maxSkills = 0;
+let userWithMostSkills = null;
+
+for (const username in users) {
+    const userSkillsCount = users[username].skills.length;
+    if (userSkillsCount > maxSkills) {
+        maxSkills = userSkillsCount; 
+        userWithMostSkills = username;  
+    }
+}
+console.log(`${userWithMostSkills} with ${maxSkills} skills` );
+
+
+// Count logged in users, count users having greater than equal to 50 points from the following object. 
+let loggedInCount = 0;
+let highPointsCount = 0;
+
+for (const username in users) {
+    const user = users[username];
+    if (user.isLoggedIn) {
+        loggedInCount++; 
+    }
+    if (user.points >= 50) {
+        highPointsCount++;  
+    }
+}
+
+console.log(`Logged in users: ${loggedInCount}`);
+console.log(`Users with >= 50 points: ${highPointsCount}`);
+
+
+// Find people who are MERN stack developer from the users object
+const mernStack = ['MongoDB', 'Express', 'React', 'Node', 'Node.js']
+let MernDevelopers = []
+
+for (const userName in users) {
+    const userSkills = users[userName].skills   
+    if (mernStack.some(skill => userSkills.includes(skill))){
+        MernDevelopers.push(userName)
+    }
+}
+// console.log(MernDevelopers);
+
+// Set your name in the users object without modifying the original users object
+const newUsers = Object.assign({},users)
+newUsers['Dmitry'] = {      
+    email: 'dmtlazarev@gmail.com',
+    skills: ['HTML', 'CSS', 'JavaScript'],
+    age: 44,
+    isLoggedIn: false,
+    points: 50
+};
+// console.log(users, newUsers)
+
+// Get all keys or properties of users object
+const keys = Object.keys(newUsers)
+
+// Get all the values of users object
+const values = Object.values(newUsers)
+
+
+// Exercises: Level 3
+
+// Create an object literal called personAccount. 
+// It has firstName, lastName, incomes, expenses properties and it has totalIncome, totalExpense, accountInfo,addIncome, addExpense and accountBalance methods. 
+// Incomes is a set of incomes and its description and expenses is a set of incomes and its description.
+const personAccount = {
+    firstName: 'John',
+    lastName: 'Smith',
+    incomes: [
+        { description: 'Salary', amount: 3000 },
+        { description: 'Freelance', amount: 1500 }
+    ],
+    expenses: [
+        { description: 'Food', amount: 300 },
+        { description: 'Rent', amount: 1200 },
+        { description: 'Other expenses', amount: 300 }
+    ],
+   
+    totalIncome() {
```

Comment:
According to naming conventions, a method name should be a verb and describe the action, e.g. `getTotalIncome`, `calculcateTotalIncome`.

Reply:
Updated